crates/router/src/connector/payeezy/transformers.rs

```diff
@@ -0,0 +1,498 @@
+pub struct PayeezyPaymentsResponse {
+    correlation_id: String,
+    transaction_status: PayeezyPaymentStatus,
+    validation_status: String,
+    transaction_type: PayeezyTransactionType,
+    transaction_id: String,
+    transaction_tag: Option<String>,
+    method: Option<String>,
+    amount: String,
+    currency: String,
+    bank_resp_code: String,
+    bank_message: String,
+    gateway_resp_code: String,
+    gateway_message: String,
+    stored_credentials: Option<PaymentsStoredCredentials>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PaymentsStoredCredentials {
+    cardbrand_original_transaction_id: String,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayeezyCaptureOrVoidRequest {
+    transaction_tag: String,
+    transaction_type: PayeezyTransactionType,
+    amount: String,
+    currency_code: String,
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for PayeezyCaptureOrVoidRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        let payment_details = item
+            .request
+            .connector_meta
+            .as_ref()
+            .get_required_value("connector_meta")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "connector_meta",
+            })?
+            .clone();
+        let metadata: PayeezyPaymentsMetadata = payment_details
+            .parse_value("PayeezyPaymentsMetadata")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "transaction_tag",
+            })?;
+        Ok(Self {
+            transaction_type: PayeezyTransactionType::Capture,
+            amount: item.request.amount.to_string(),
+            currency_code: item.request.currency.to_string(),
+            transaction_tag: metadata.transaction_tag,
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsCancelRouterData> for PayeezyCaptureOrVoidRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        let payment_details = item
+            .request
+            .connector_meta
+            .as_ref()
+            .get_required_value("connector_meta")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "connector_meta",
+            })?
+            .clone();
+        let metadata: PayeezyPaymentsMetadata = payment_details
+            .parse_value("PayeezyPaymentsMetadata")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "transaction_tag",
+            })?;
+        Ok(Self {
+            transaction_type: PayeezyTransactionType::Void,
+            amount: item.request.amount.unwrap_or_default().to_string(),
```

Comment:
`unwrap_or_default`? in case if amount is None then default value of i64 will be 0. 

If amount is mandatory then throw proper error or else keep amount as Option